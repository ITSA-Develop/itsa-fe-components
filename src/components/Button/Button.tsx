import { Button as AntButton, ButtonProps } from 'antd';

export const Button = ({ children, ...rest }: ButtonProps) => {
	return <AntButton {...rest}>{children}</AntButton>;
};
