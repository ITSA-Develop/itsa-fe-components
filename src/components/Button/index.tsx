import { ButtonAntd } from '../ButtonAntd';
import { ReactNode } from 'react';

export interface IButtonProps {
	size?: 'small' | 'middle' | 'large';
	type?: 'primary' | 'secondary';
	title?: ReactNode;
	disabled?: boolean;
	onClick?: () => void;
	/** Cuando es true en tipo secondary, aplica estilo gris (#595959) */
	default?: boolean;
}

export const Button = (props: IButtonProps) => {
	const { size = 'small', type = 'primary', title, disabled = false, onClick } = props;
	const sizeClass = size === 'small' ? 'itsa-btn--sm' : size === 'middle' ? 'itsa-btn--md' : 'itsa-btn--lg';
	const variantClass = type === 'primary' ? 'itsa-btn--primary' : 'itsa-btn--secondary';
	const disabledClass = disabled ? 'itsa-btn--disabled' : '';
	const defaultSecondaryClass = type === 'secondary' && props.default ? 'itsa-btn--default' : '';
	const className = ['itsa-btn', sizeClass, variantClass, defaultSecondaryClass, disabledClass].filter(Boolean).join(' ');
	const antdType: 'primary' | 'default' = type === 'primary' ? 'primary' : 'default';

	return (
		<ButtonAntd className={className} size={size} type={antdType} disabled={disabled} onClick={onClick}>
			{title}
		</ButtonAntd>
	);
};
