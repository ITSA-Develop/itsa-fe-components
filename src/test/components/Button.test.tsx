import { fireEvent, render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import { Button } from '../../components/Button/Button';

describe('Button component', () => {
	it('renders children text', () => {
		const { container } = render(<Button>Click me</Button>);
		expect(screen.getByText('Click me')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('passes props to AntButton', () => {
		const { container } = render(<Button type="primary">Primary</Button>);
		const button = screen.getByRole('button');
		expect(button).toHaveClass('ant-btn-primary');
		expect(container).toMatchSnapshot();
	});

	it('calls onClick handler when clicked', () => {
		const handleClick = vi.fn();
		const { container } = render(<Button onClick={handleClick}>Click</Button>);
		const button = screen.getByRole('button');
		fireEvent.click(button);
		expect(handleClick).toHaveBeenCalledTimes(1);
		expect(container).toMatchSnapshot();
	});

	it('renders disabled button', () => {
		const { container } = render(<Button disabled>Disabled</Button>);
		const button = screen.getByRole('button');
		expect(button).toBeDisabled();
		expect(container).toMatchSnapshot();
	});
});
