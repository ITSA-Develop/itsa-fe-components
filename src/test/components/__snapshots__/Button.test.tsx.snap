// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > calls onClick handler when clicked 1`] = `
<div>
  <button
    class="ant-btn css-dev-only-do-not-override-l9pxc0 ant-btn-default ant-btn-color-default ant-btn-variant-outlined"
    type="button"
  >
    <span>
      Click
    </span>
  </button>
</div>
`;

exports[`Button component > passes props to AntButton 1`] = `
<div>
  <button
    class="ant-btn css-dev-only-do-not-override-l9pxc0 ant-btn-primary ant-btn-color-primary ant-btn-variant-solid"
    type="button"
  >
    <span>
      Primary
    </span>
  </button>
</div>
`;

exports[`Button component > renders children text 1`] = `
<div>
  <button
    class="ant-btn css-dev-only-do-not-override-l9pxc0 ant-btn-default ant-btn-color-default ant-btn-variant-outlined"
    type="button"
  >
    <span>
      Click me
    </span>
  </button>
</div>
`;

exports[`Button component > renders disabled button 1`] = `
<div>
  <button
    class="ant-btn css-dev-only-do-not-override-l9pxc0 ant-btn-default ant-btn-color-default ant-btn-variant-outlined"
    disabled=""
    type="button"
  >
    <span>
      Disabled
    </span>
  </button>
</div>
`;
